xbmc/utils/DateLib.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ *  Copyright (C) 2005-2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "DateLib.h"
+
+#include "utils/XTimeUtils.h"
+
+#include <cstdlib>
+
+const std::string KODI::TIME::ExtractTzName(const char* tzname)
+{
+  if (tzname[0] != '/')
+    return tzname;
+
+  std::string result = tzname;
+  const char zoneinfo[] = "zoneinfo";
+  size_t pos = result.rfind(zoneinfo);
+  if (pos == result.npos) {
+    return "";
+  }
+
+  pos = result.find('/', pos);
+  result.erase(0, pos + 1);
+  return result;
+}
+
+const date::time_zone* KODI::TIME::GetTimeZone()
+{
+#if defined(TARGET_POSIX) && !defined(TARGET_DARWIN)
+  const date::time_zone* tz = nullptr;
+  const char *tzname = std::getenv("TZ");
+
+  if (tzname && tzname[0] != '\0')
+  {
+    const std::string tzn = ExtractTzName(tzname);
+    if (!tzn.empty())
+      tz = date::locate_zone(tzn);
```

Comment:
This crashes in the unit tests, probably `date::set_install(tzdataPath);` wasn't called before? Trace:
```
terminate called after throwing an instance of 'std::runtime_error'
  what():  Timezone database not found at "/home/mark/Downloads/tzdata"

Thread 272 "MHD-connection" received signal SIGABRT, Aborted.
[Switching to Thread 0x7fffc04906c0 (LWP 173537)]
__pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at pthread_kill.c:44
44            return INTERNAL_SYSCALL_ERROR_P (ret) ? INTERNAL_SYSCALL_ERRNO (ret) : 0;
(gdb) bt
#0  __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at pthread_kill.c:44
#1  0x00007ffff6e176b3 in __pthread_kill_internal (signo=6, threadid=<optimized out>) at pthread_kill.c:78
#2  0x00007ffff6dc7958 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#3  0x00007ffff6db153d in __GI_abort () at abort.c:79
#4  0x00007ffff5699833 in __gnu_cxx::__verbose_terminate_handler () at /usr/src/debug/gcc/libstdc++-v3/libsupc++/vterminate.cc:95
#5  0x00007ffff56a5cfc in __cxxabiv1::__terminate (handler=<optimized out>) at /usr/src/debug/gcc/libstdc++-v3/libsupc++/eh_terminate.cc:48
#6  0x00007ffff56a5d69 in std::terminate () at /usr/src/debug/gcc/libstdc++-v3/libsupc++/eh_terminate.cc:58
#7  0x00007ffff56a5fcd in __cxxabiv1::__cxa_throw (obj=<optimized out>, tinfo=0x7ffff58291e0 <typeinfo for std::runtime_error>, dest=0x7ffff56bd470 <std::runtime_error::~runtime_error()>) at /usr/src/debug/gcc/libstdc++-v3/libsupc++/eh_throw.cc:98
#8  0x0000555567c905ee in date::init_tzdb() ()
#9  0x0000555567c86f2f in date::create_tzdb() ()
#10 0x0000555567c86feb in date::get_tzdb_list() ()
#11 0x0000555567c90e88 in date::get_tzdb() ()
#12 0x0000555567c92605 in date::current_zone() ()
#13 0x000055556226c8c6 in KODI::TIME::GetTimeZone () at /home/mark/Coding/Repos/kodi-git/xbmc/xbmc/utils/DateLib.cpp:49
#14 0x0000555564b3c8b7 in CDateTime::GetCurrentDateTime () at /home/mark/Coding/Repos/kodi-git/xbmc/xbmc/XBDateTime.cpp:207
#15 0x0000555566386062 in CWebServer::FinalizeRequest (this=0x61a00007b690, handler=warning: RTTI symbol not found for class 'std::_Sp_counted_ptr<IHTTPRequestHandler*, (__gnu_cxx::_Lock_policy)2>'
warning: RTTI symbol not found for class 'std::_Sp_counted_ptr<IHTTPRequestHandler*, (__gnu_cxx::_Lock_policy)2>'
std::shared_ptr<IHTTPRequestHandler> (use count 1, weak count 0) = {...}, responseStatus=200, response=0x61d000071c80)
    at /home/mark/Coding/Repos/kodi-git/xbmc/xbmc/network/WebServer.cpp:427
#16 0x000055556637c00d in CWebServer::HandleRequest (this=0x61a00007b690, handler=warning: RTTI symbol not found for class 'std::_Sp_counted_ptr<IHTTPRequestHandler*, (__gnu_cxx::_Lock_policy)2>'
warning: RTTI symbol not found for class 'std::_Sp_counted_ptr<IHTTPRequestHandler*, (__gnu_cxx::_Lock_policy)2>'
std::shared_ptr<IHTTPRequestHandler> (use count 1, weak count 0) = {...}) at /home/mark/Coding/Repos/kodi-git/xbmc/xbmc/network/WebServer.cpp:375
#17 0x0000555566373536 in CWebServer::HandlePartialRequest (this=0x61a00007b690, connection=0x615000171400, connectionHandler=0x607000c24f70, request=..., upload_data=0x0, upload_data_size=0x7fffc048f640, con_cls=0x615000171458)
    at /home/mark/Coding/Repos/kodi-git/xbmc/xbmc/network/WebServer.cpp:259
#18 0x000055556636f194 in CWebServer::AnswerToConnection (cls=0x61a00007b690, connection=0x615000171400, url=0x62d000168404 "/vfs//home/mark/Coding/Repos/kodi-git/build/xbmc/network/test/data/webserver/test-ranges.txt", method=0x62d000168400 "GET", 
    version=0x62d000168479 "HTTP/1.1", upload_data=0x0, upload_data_size=0x7fffc048f640, con_cls=0x615000171458) at /home/mark/Coding/Repos/kodi-git/xbmc/xbmc/network/WebServer.cpp:174
#19 0x00007ffff7a07d5c in ?? () from /usr/lib/libmicrohttpd.so.12
#20 0x00007ffff7a0a248 in ?? () from /usr/lib/libmicrohttpd.so.12
#21 0x00007ffff7a0c770 in ?? () from /usr/lib/libmicrohttpd.so.12
#22 0x00007ffff7a0db0d in ?? () from /usr/lib/libmicrohttpd.so.12
#23 0x00007ffff6e158fd in start_thread (arg=<optimized out>) at pthread_create.c:442
#24 0x00007ffff6e97a60 in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81

```